data-prepper-plugins/mutate-event-processors/README.md

```diff
@@ -246,6 +246,101 @@ and the type conversion processor will change it to the following output, where
+
+### Basic Usage
+
+To get started with list-to-map processor using Data Prepper, create the following `pipeline.yaml`:
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: true
+  sink:
+    - stdout:
+```
+
+Create the file named `logs_json.log` with the following content and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}]}
+```
+
+When run, the processor will parse the message into the following output:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":"val-a","b":"val-b1","c":"val-c"}
+```
+
+If we set a `target`:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        target: "mymap"
+        value_key: "value"
+        flatten: true
+```
+the generated map will be put under the target key:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"mymap":{"a":"val-a","b":"val-b1","c":"val-c"}}
+```
+
+If we do not specify a `value_key`:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        flatten: true
+```
+the values of the map will be original objects from the source list:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":{"name":"a","value":"val-a"},"b":{"name":"b","value":"val-b1"},"c":{"name":"c","value":"val-c"}}
+```
+
+If `flatten` option is not set or set to false:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: false
+```
+the values of the map will be lists, and some entries may have more than one element in their values:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":["val-a"],"b":["val-b1","val-b2"],"c":["val-c"]}
+```
+
+If we specify `flattened_element` as "last":
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: true
+        flattened_element: "last"
+```
+the last element will be kept:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":"val-a","b":"val-b2","c":"val-c"}
+```
+
+### Configuration
+* `key` - (required) - The key of the fields that will be extracted as keys in the generated map
+* `source` - (required) - The key of the source list
```

Comment:
The description is not clear enough here. 
Maybe something like, The key in the event with list of objects that will be converted to map.

Reply:
Thanks! Updated.